### DISCUSSION TITLE
How to use hardware acceleration in gstreamer in termux?

### DISCUSSION THREAD

**Discussion by xaionaro** [termux/termux-packages#20293] [Q&A]
Hello.

I currently use `hevc_mediacodec` encoder in `ffmpeg` and I want to switch to `gstreamer`. How to use hardware acceleration for HEVC encoding using `gstreamer` in termux? Could anybody provide a simple example of a pipeline?

**Comment by wszqkzqk** [comment]
> Hello.
> 
> I currently use `hevc_mediacodec` encoder in `ffmpeg` and I want to switch to `gstreamer`. How to use hardware acceleration for HEVC encoding using `gstreamer` in termux? Could anybody provide a simple example of a pipeline?

How did you use `hevc_mediacodec` in your ffmpeg, could you please provide your CLI?

**Reply by Randrianasulu** [reply]
> > Hello.
> > I currently use `hevc_mediacodec` encoder in `ffmpeg` and I want to switch to `gstreamer`. How to use hardware acceleration for HEVC encoding using `gstreamer` in termux? Could anybody provide a simple example of a pipeline?
> 
> How did you use `hevc_mediacodec` in your ffmpeg, could you please provide your CLI?

```
./ffmpeg -i ~/AMV\ -\ Ghost\ in\ the\ Shell\ \(1995\)\ -\ Inner\ Universe\ \[Bd2LThO0PcQ].webm -c:v hevc_mediacodec -b:v 15M -c:a copy  -pix_fmt nv12  test5.mp4

./ffmpeg -c:v vp9_mediacodec -i ~/AMV\ -\ Ghost\ in\ the\ Shell\ \(1995\)\ -\ Inner\ Universe\ \[Bd2LThO0PcQ].webm -c:v hevc_mediacodec -vf drawbox  -b:v 1M -c:a copy  -pix_fmt nv12  test7.mp4

```

two examples I saved on Termux
